source/administration/config-settings.rst

```diff
@@ -361,6 +361,16 @@ A 32-character key for encrypting and decrypting sensitive fields in the databas
 
 When using High Availability, the salt must be identical in each instance of Mattermost.
 
+The following fields are encrypted using this key
+
+- ``SqlSettings.DriverName``
+- ``SqlSettings.DataSource``
+- ``SqlSettings.MaxIdleConns``
+- ``SqlSettings.MaxOpenConns``
+- ``SqlSettings.Trace``
+- ``SqlSettings.QueryTimeout``
+- ``SqlSettings.ConnMaxLifetimeMilliseconds``
```

Comment:
I'm pretty sure this isn't true. I don't think the AtRestEncrypt key is used for anything at the moment and is a legacy setting. I'm curious as to how you came up with that?

Reply:
I found `AtRestEncrypt` https://github.com/search?p=2&q=org%3Amattermost+SqlSettings&type=Code being set at quite a few places, I think the fields of `SqlSettings` are being encrypted using `AtRestEncrypt`, right?

Reply:
See this https://github.com/search?q=org%3Amattermost+SqlSettings.AtRestEncryptKey&type=Code

Reply:
Sure there are reference to it in code but it's not being used for any actual encrypting. Can you point to where it's actually doing what you say?

>The following fields are encrypted using this key

Reply:
I looked at quite a lot of things, I don't remember now, correct if I'm wrong.
- https://github.com/mattermost/mattermost-server/commit/34d688ca72d8ef13b83987cc3b3d9b93ab989d9f
- https://github.com/mattermost/docs/issues/2231
- https://github.com/mattermost/mattermost-server/issues/9741
- https://github.com/mattermost/mattermost-redhat-docker/blob/e9d707231cc867fcc265fdb3ebe4073c3e91a340/v5/entrypoint.sh#L52

I didn't find a direct reference of `AtRestEncryptKey`, what I understood was `AtRestEncryptKey` encrypts anything `SqlSettings`. I agree that there is no need for encrypting all fields of `SqlSettings`. Why do you have `AtRestEncryptKey` then? Just curious.

Reply:
It's a legacy setting that we used to use for encrypting data being stored at rest in the database. The setting was never really fully used and we just haven't removed it

Reply:
If you want to remove this section from the PR the other parts look good

Reply:
I added a31109e, looks ok? or should I remove this line? I suggest you create a ticket for this.

Reply:
Does a31109e, looks ok? or should I remove the line? I suggest you create a ticket.

Reply:
That's fine, thanks!